lib/mail_room/structured_logger.rb

```diff
@@ -0,0 +1,21 @@
+require 'logger'
+require 'json'
+
+module MailRoom
+  class StructuredLogger < Logger
+
+    def format_message(severity, timestamp, progname, message)
+      data = {}
+      data[:severity] = severity
+      data[:time] = timestamp || Time.now.to_s
+
+      if message.is_a? String
```

Comment:
Understand the need to implement this, but I'd recommend to input hashes only in this PR. That would allow for better indexing in Splunk/ELK/Loki/etc. Text indexing is much less structured.